File: src/main/java/no/jstien/bikeapi/tsdb/write/Datum.java
package no.jstien.bikeapi.tsdb.write;

import java.util.HashMap;
import java.util.Map;

class Datum {
    private String metric;
    private long value;
    private long timestamp;
    private Map<String,String> tags = new HashMap<>();

    Datum(String metric, long value, long timestamp) {
        this.metric = metric;
        this.value = value;
        this.timestamp = timestamp;
    }

    Datum(String metric, long value) {
        this.metric = metric;
        this.value = value;
        this.timestamp = System.currentTimeMillis() / 1000L;
    }

    public void addTag(String key, String value) {
        tags.put(key, value);
    }

    public String getMetric() {
        return metric;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public long getValue() {
        return value;
    }


    @Override
    public String toString() {
        return String.format("{metric:'%s', value:'%d', timestamp:'%d' tags:'%s'", metric, value, timestamp, tags.toString());
    }
}

File: src/main/java/no/jstien/bikeapi/tsdb/write/Datum.kt
package no.jstien.bikeapi.tsdb.write

import java.util.HashMap

class Datum {
    var metric: String? = null
        private set
    var value: Long = 0
        private set
    var timestamp: Long = 0
        private set
    private val tags = HashMap<String, String>()

    constructor(metric: String, value: Long, timestamp: Long) {
        this.metric = metric
        this.value = value
        this.timestamp = timestamp
    }

    constructor(metric: String, value: Long) {
        this.metric = metric
        this.value = value
        this.timestamp = System.currentTimeMillis() / 1000L
    }

    fun addTag(key: String, value: String) {
        tags[key] = value
    }


    override fun toString(): String {
        return String.format("{metric:'%s', value:'%d', timestamp:'%d' tags:'%s'", metric, value, timestamp, tags.toString())
    }
}

